### DISCUSSION TITLE
Dynamically setting the ForceCulture attribute

### DISCUSSION THREAD

**Discussion by belugabob** [SeriousM/WPFLocalizationExtension#357] [Q&A] [answered]
I have a need to be able to bind the value that I assign to the ForceCulture attribute, so that I can display translations into different languages at different places on the screen.

I currently have a ListView that is bound to a list of language codes and wish to display a phrase in English and the relevant language, for each language in the list.

If I do this, then the language column displays all of the language codes, but the other columns always display the french translations

``` xml
   <Window.Resources>
        <DataTemplate x:Key="englishNameOfLanguageCellTemplate">
            <TextBlock Text="{lex:LocText Key='EnglishNameOfLanguage', ForceCulture=fr'}"/>
        </DataTemplate>
        <DataTemplate x:Key="localNameOfLanguageCellTemplate">
            <TextBlock Text="{lex:LocText Key='LocalNameOfLanguage', ForceCulture=fr}"/>
        </DataTemplate>
    </Window.Resources>
            <ListView ItemsSource="{Binding Codes}" Margin="10,36,703,357">
                <ListView.View>
                    <GridView>
                        <GridViewColumn Header="Language" Width="auto" DisplayMemberBinding="{Binding}" />
                        <GridViewColumn Header="English name" Width="auto" CellTemplate="{StaticResource englishNameOfLanguageCellTemplate}" />
                        <GridViewColumn Header="Local name" Width="auto" CellTemplate="{StaticResource localNameOfLanguageCellTemplate}" />
                    </GridView>
                </ListView.View>
            </ListView>
```

What I'd hoped to be able to do is...

`            <TextBlock Text="{lex:LocText Key='EnglishNameOfLanguage', ForceCulture={Binding}'}"/>
`
`            <TextBlock Text="{lex:LocText Key='LocalNameOfLanguage', ForceCulture={Binding}}"/>
`
...in order to get the translation for the appropriate language.

Unfortunately this produces the error...

`Object of type 'System.Windows.Data.Binding' cannot be converted to type 'System.String'`

Is there a way of dynamically setting the ForceCulture attribute?

**Comment by Karnah** [comment, marked as answer]
Hello, @belugabob!
Unfortunately, it's really very hard to replace `ForceCulture` from `string` to `Binding`. I'll have some ideas how to make it's more possible.

But I have one idea how you can deal with it. You can write converter like this:
``` c#
public class CultureNameConverter : TypeValueConverterBase, IValueConverter
{
    /// <inheritdoc />
    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
    {
        var valueCulture = value as CultureInfo;
        if (valueCulture == null)
            return string.Empty;

        var key = parameter as string;
        if (string.IsNullOrEmpty(key))
            return string.Empty;

        return Ressourcen.ResourceManager.GetObject(key, valueCulture);
    }

    /// <inheritdoc />
    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
    {
        throw new NotSupportedException();
    }
}
```

and using:
``` xaml
<TextBlock Text="{Binding Converter={c:CultureNameConverter}, ConverterParameter='EnglishNameOfLanguage'}" />
<TextBlock Text="{Binding Converter={c:CultureNameConverter}, ConverterParameter='LocalNameOfLanguage'}" />
```

So instead of `Ressourcen.ResourceManager` you should use your own dictionary. You could also try to use `LocalizeDictionary.Instance.GetLocalizedObject(key, null, valueCulture)`, but you can have a problem with `Provider` if you don't use a `Default`.

I don't test it, but it should work :)

**Reply by belugabob** [reply]
You, Sir, are a star.

The thing that I was missing, was spinning the key and culture code around, so that the culture code could be bound to and leaving the key as a parameter, as it is constant - obvious, when you see it.

Much appreciated.
